File: Day 8/python/ceaser_cipher.py
# This was my method in solving this problem.

# Does not account for large shift amounts

alphabet = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']

direction = input("Type 'encode' to encrypt, type 'decode' to decrypt:\n")
text = input("Type your message:\n").lower()
shift = int(input("Type the shift number:\n"))

def ceaser(start_text,shift_amount,cipher_direction):
    found_text = ""
    for letter in start_text:
        position = alphabet.index(letter)
        if cipher_direction == 'encode':
            new_position = position + shift_amount
            if new_position > 25:
                new_position -= 26
            found_text += alphabet[new_position]
        else:
            new_position = position - shift_amount
            if new_position < 0:
                new_position += 26
            found_text += alphabet[new_position]
    
    print(f"The {cipher_direction}d text is {found_text}")

ceaser(start_text=text,shift_amount=shift,cipher_direction=direction)

File: Day 8/javascript/ceaser_cipher.js
const alphabet = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'];

function caesar(start_text, shift_amount, cipher_direction) {
    let end_text = "";

    if (cipher_direction === "decode") {
        shift_amount *= -1;
    };

    for (let char = 0; char < start_text.lenght; char++) {
        if (alphabet.includes(start_text[char])) {
            let position = alphabet.indexOf(start_text[char]);
            let new_position = position + shift_amount;
            end_text += alphabet[new_position]
        } else {
            end_text += start_text[char];
        };
    };
    console.log(end_text)
    console.log("Here's the " + cipher_direction +"d result: " + end_text);
};

// Import and print the logo from art.py when the program starts.
// import art
// print(art.logo)

let cipher = true;

while (cipher) {
   

    let direction = prompt("Type 'encode' to encrypt, type 'decode' to decrypt:\n");
    let text = prompt("Type your message:\n").toLowerCase();
    let shift = Number(prompt("Type the shift number:\n"));

    shift = shift % 26;
  

    caesar(start_text=text, shift_amount=shift, cipher_direction=direction);

    let proceed = prompt("Type 'yes' if you want to go again. Otherwise type 'no'.\n");

    if (proceed === 'no') {
        cipher = false;
        console.log('Goodbye!');
    };     
};